src/main/java/services/DatafeedEventsServiceV2.java

```diff
@@ -29,33 +29,34 @@ public DatafeedEventsServiceV2(SymBotClient client) {
      */
     @Override
     public void readDatafeed() {
-        try {
-            String datafeedId;
-            List<DatafeedV2> datafeedIds = datafeedClient.listDatafeedId();
-            if (datafeedIds.isEmpty()) {
-                datafeedId = this.createDatafeedId();
-            } else {
-                //Each bot should subscribe only one datafeed
-                datafeedId = datafeedIds.get(0).getId();
-            }
-
-            //Read datafeed in loop
-            started.set(true);
-            List<DatafeedEvent> events = null;
-            do {
-                events = datafeedClient.readDatafeed(datafeedId, datafeedClient.getAckId());
-                if (events != null && !events.isEmpty()) {
-                    handleEvents(events);
+        new Thread(() -> {
+            try {
+                String datafeedId;
+                List<DatafeedV2> datafeedIds = datafeedClient.listDatafeedId();
+                if (datafeedIds.isEmpty()) {
+                    datafeedId = this.createDatafeedId();
+                } else {
+                    //Each bot should subscribe only one datafeed
+                    datafeedId = datafeedIds.get(0).getId();
                 }
-            } while (started.get());
-        } catch (Exception e) {
-            this.started.set(false);
-            logger.error("Something went wrong while reading datafeed", e);
-            logger.info("Sleeping for {} seconds before retrying..", botClient.getConfig().getDatafeedEventsErrorTimeout());
-            sleep(botClient.getConfig().getDatafeedEventsErrorTimeout());
-            readDatafeed();
-        }
 
+                //Read datafeed in loop
+                started.set(true);
+                List<DatafeedEvent> events = null;
+                do {
+                    events = datafeedClient.readDatafeed(datafeedId, datafeedClient.getAckId());
+                    if (events != null && !events.isEmpty()) {
+                        handleEvents(events);
+                    }
+                } while (started.get());
+            } catch (Exception e) {
+                this.started.set(false);
+                logger.error("Something went wrong while reading datafeed", e);
+                logger.info("Sleeping for {} seconds before retrying..", botClient.getConfig().getDatafeedEventsErrorTimeout());
+                sleep(botClient.getConfig().getDatafeedEventsErrorTimeout());
+                readDatafeed();
```

Comment:
aren't we going to spawn a thread from a thread each time there is an exception?